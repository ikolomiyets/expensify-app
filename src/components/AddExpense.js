import React from 'react';

const AddExpense = () => (
    <div className="container">
        <div className="h1">React JS Starter Project</div>
        <p>This is my add expense page</p>
    </div>
);

export default AddExpense;