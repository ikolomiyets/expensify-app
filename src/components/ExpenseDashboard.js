import React from 'react';

const ExpenseDashboard = () => (
    <div className="container">
        <div className="h1">React JS Starter Project</div>
        <p>Font Awesome Icon <i className="fa fa-snowflake-o"></i></p>
    </div>
);

export default ExpenseDashboard;