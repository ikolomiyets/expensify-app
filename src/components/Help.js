import React from 'react';

const Help = () => (
    <div className="container">
        <div className="h1">React JS Starter Project</div>
        <p>This is my help page</p>
    </div>
);

export default Help;