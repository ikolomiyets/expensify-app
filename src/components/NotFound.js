import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div className="container">
        404! - <Link to="/">Go Back</Link>
    </div>
);

export default NotFoundPage;