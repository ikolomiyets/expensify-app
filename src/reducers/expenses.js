import * as fromActions from '../actions/expenses';

const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
    if (action) {
        switch(action.type) {
            case fromActions.ADD_EXPENSE:
                return  [
                    ...state,
                    action.expense
                ];
            case fromActions.REMOVE_EXPENSE:
                return state.filter(({ id }) => id !== action.id);
            case fromActions.EDIT_EXPENSE:
                return state.map((expense) => {
                    if (expense.id === action.id) {
                        return {
                            ...expense,
                            ...action.update
                        }
                    } else {
                        return expense;
                    }
                });
            default:
                return state;
        }
    } else {
        return state;
    }
};

