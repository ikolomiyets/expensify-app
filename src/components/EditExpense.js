import React from 'react';
import ExpenseForm from './ExpenseForm';
import { editExpense } from '../actions/expenses';
import { connect } from 'react-redux';

export class EditExpense extends React.Component {
    constructor(props) {
        super(props);
    }

    onSubmit = (expense) => {
        this.props.onSubmit(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    render() {
        return (
            <div className="container">
                <h1>Edit Expense</h1>
                <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit}/>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id;
        })
    };
};

const mapDispatchToProps = (dispatch) => ({
    onSubmit: (expense) => dispatch(editExpense(id, expense))
});


export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);