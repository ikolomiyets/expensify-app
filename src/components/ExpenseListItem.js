import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from "../actions/expenses";
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt, dispatch} ) => (
    <p><Link to={`/edit/${id}`}>{description}</Link> - {amount} - {createdAt}<button onClick={() => {
       dispatch(removeExpense(id));
   }}>Remove</button></p>
);

export default connect()(ExpenseListItem);