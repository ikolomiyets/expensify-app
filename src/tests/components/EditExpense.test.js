import React from 'react';
import { shallow } from 'enzyme';
import ExpenseForm from '../../components/ExpenseForm';
import expenses from '../fixtures/expenses';
import { EditExpense } from '../../components/EditExpense';

let onSubmitSpy, historySpy, wrapper;

beforeEach(() => {
    onSubmitSpy = jest.fn();
    historySpy = { push: jest.fn() };

    wrapper = shallow(<EditExpense expense={expenses[2]} onSubmit={onSubmitSpy} history={historySpy}/>);
});

test('should render EditExpense page correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
    wrapper.find(ExpenseForm).prop('onSubmit')(expenses[1]);
    expect(historySpy.push).toHaveBeenLastCalledWith('/');
    expect(onSubmitSpy).toHaveBeenLastCalledWith(expenses[2].id, expenses[1]);
});

