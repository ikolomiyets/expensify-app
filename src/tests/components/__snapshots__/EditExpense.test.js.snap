// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpense page correctly 1`] = `
<div
  className="container"
>
  <h1>
    Edit Expense
  </h1>
  <ExpenseForm
    expense={
      Object {
        "amount": 4500,
        "createdAt": 345600000,
        "description": "Credit card",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
</div>
`;
